Rakefile

```diff
@@ -23,38 +24,106 @@ task :validate do
   end
 end
 
-task :acceptance do
-  require 'beaker-hostgenerator'
+namespace :litmus do
+  # Install the puppet module fixture on a collection of nodes
+  #
+  # @param :target_node_name [Array] nodes on which to install a puppet module for testing.
+  desc 'install_module_fixtures - build and install module fixtures'
+  task :install_module_fixtures, [:target_node_name] do |_task, args|
+    inventory_hash = inventory_hash_from_inventory_file
+    target_nodes = find_targets(inventory_hash, args[:target_node_name])
+    if target_nodes.empty?
+      puts 'No targets found'
+      exit 0
+    end
+    include BoltSpec::Run
+    require 'pdk/module/build'
 
-  install_type = 'aio'
-  target = ENV['platform']
-  abs = if ENV['BEAKER_ABS'] then 'abs' else 'vmpooler' end
-  if ! target
-    STDERR.puts 'TEST_TARGET environment variable is not set'
-    STDERR.puts 'setting to default value of "centos7-64ma".'
-    target = "centos7-64ma{type=#{install_type}}"
-  end
+    module_fixture_dir = File.expand_path(File.join(File.dirname(__FILE__), 'spec', 'fixtures', 'acceptance', 'modules', 'test'))
+    module_tar = nil
+    Dir.chdir(module_fixture_dir) do
+      opts = {}
+      opts[:force] = true
+      builder = PDK::Module::Build.new(opts)
+      module_tar = builder.build
+      puts 'Built'
+      module_tar = Dir.glob('pkg/*.tar.gz').max_by { |f| File.mtime(f) }
+      raise "Unable to find package in 'pkg/*.tar.gz'" if module_tar.nil?
+      module_tar = File.expand_path(module_tar)
+    end
 
-  unless target =~ /type=/
-    puts "INFO: adding 'type=#{install_type}' to host config"
-    target += "{type=#{install_type}}"
+    target_string = if args[:target_node_name].nil?
+                      'all'
+                    else
+                      args[:target_node_name]
+                    end
+    # TODO: Currently this is Linux only
+    tmp_path = '/tmp/'
+    run_local_command("bundle exec bolt file upload #{module_tar} #{tmp_path}#{File.basename(module_tar)} --nodes #{target_string} --inventoryfile inventory.yaml")
```

Comment:
SHould this be using the BOlt:Runner namespace stuff?

Reply:
It probably should but .. the current litmus rake helpers use this style e.g.

https://github.com/puppetlabs/puppet_litmus/blob/45feb78c2df91b2110900e63baf89693e9dbd511/lib/puppet_litmus/rake_tasks.rb#L283